### ISSUE TITLE
Optionally return transport plans for the Sinkhorn loss

### ISSUE THREAD

**Issue by arnaudvl** [jeanfeydy/geomloss#61]
First of all, thanks for the great package (incl. KeOps). I believe it could be very helpful for the user to be able to inspect the optimal transport plans found by the Sinkhorn iterations. This could e.g. be done by optionally computing them after the ```sinkhorn_loop``` routine following Eq.9 of [Interpolating between Optimal Transport and MMD using Sinkhorn Divergences
](https://arxiv.org/abs/1810.08278) (simplified for batch size of 1):

```python
p_xy = (((1 / ε) * (f_ba.unsqueeze(-1) + g_ab.unsqueeze(0) - C_xy[0])).exp() * (α.unsqueeze(-1) @ β.unsqueeze(0)))[0]
p_xx, p_yy = None, None
if debias:
  p_xx = (((1 / ε) * (f_aa.unsqueeze(-1) + f_aa.unsqueeze(0) - C_xx[0])).exp() * (α.unsqueeze(-1) @ α.unsqueeze(0)))[0]
  p_yy = (((1 / ε) * (g_bb.unsqueeze(-1) + g_bb.unsqueeze(0) - C_yy[0])).exp() * (β.unsqueeze(-1) @ β.unsqueeze(0)))[0]
```
Thanks,
Arnaud

**Comment by jeanfeydy** [comment]
Hi @arnaudvl,

Thanks :-)
I fully agree with you, and have started to work on a much better API for GeomLoss in the [ot_api](https://github.com/jeanfeydy/geomloss/blob/ot_api/geomloss/ot/examples.py) branch. I will focus on this in the last two weeks of May: hopefully, I will finally be able to make a new release for GeomLoss in early June that solves this issue!

Best regards,
Jean

**Comment by arnaudvl** [comment]
Great, the examples you linked look like a very user friendly way indeed. Looking forward to the release!

**Comment by ursueugen** [comment]
Hi @jeanfeydy and thanks for the great package! 

Any plans to implement the return of the optimal transport plan?

Best regards,
Eugen

**Comment by jefferyUstc** [comment]
good idea to have the transport plan returned. Does Geomloss have the API now?
